**When analyzing the safety of neighborhoods, we have to check the criminal activity within them. After all, the people in a neighborhood are just as important as the nature that surrounds it when safety is in mind. The lower the level of crime, and the less severe they are, the safer the neighborhood is.**

In [2]:
import pandas as pd
import numpy as np

crime = pd.read_csv("CriminalActivity.csv", index_col="_id", low_memory=False)
crime.head(100)

Report_Number ReportedDate ReportedTime  Hour DayofWeek ReportedMonth  \
_id                                                                         
1    PGHP24000024   2024-01-01        00:31     0    Monday           Jan   
2    PGHP24000024   2024-01-01        00:31     0    Monday           Jan   
3    PGHP24000024   2024-01-01        00:31     0    Monday           Jan   
4    PGHP24000024   2024-01-01        00:31     0    Monday           Jan   
5    PGHP24000017   2024-01-01        00:21     0    Monday           Jan   
..            ...          ...          ...   ...       ...           ...   
96   PGHP24000496   2024-01-01        22:23    22    Monday           Jan   
97   PGHP24000496   2024-01-01        22:23    22    Monday           Jan   
98   PGHP24000497   2024-01-01        22:38    22    Monday           Jan   
99   PGHP24000497   2024-01-01        22:38    22    Monday           Jan   
100  PGHP24000497   2024-01-01        22:38    22    Monday           Jan   

                             NIBRS_Coded_Offense NIBRS_Offense_Code  \
_id                                                                   
1                         13A AGGRAVATED ASSAULT                13A   
2                               13C INTIMIDATION                13C   
3                         90Z ALL OTHER OFFENSES                90Z   
4                          23H ALL OTHER LARCENY                23H   
5    9999 Vehicle Offense (Not NIBRS Reportable)                999   
..                                           ...                ...   
96   9999 Vehicle Offense (Not NIBRS Reportable)                999   
97   9999 Vehicle Offense (Not NIBRS Reportable)                999   
98   9999 Vehicle Offense (Not NIBRS Reportable)                999   
99   9999 Vehicle Offense (Not NIBRS Reportable)                999   
100  9999 Vehicle Offense (Not NIBRS Reportable)                999   

     NIBRS_Offense_Category    NIBRS_Offense_Type NIBRS_Crime_Against  \
_id                                                                     
1          Assault Offenses    Aggravated Assault              Person   
2          Assault Offenses          Intimidation              Person   
3        All other Offenses    All other Offenses             Group B   
4    Larceny/Theft Offenses     All Other Larceny            Property   
5      Not NIBRS Reportable  Not NIBRS Reportable             Group B   
..                      ...                   ...                 ...   
96     Not NIBRS Reportable  Not NIBRS Reportable             Group B   
97     Not NIBRS Reportable  Not NIBRS Reportable             Group B   
98     Not NIBRS Reportable  Not NIBRS Reportable             Group B   
99     Not NIBRS Reportable  Not NIBRS Reportable             Group B   
100    Not NIBRS Reportable  Not NIBRS Reportable             Group B   

    NIBRS_Offense_Grouping                                          Violation  \
_id                                                                             
1                        A          18 2718 A1 Strangulation Basic - Applying   
2                        A             18 2706 A1 Terroristic Threats-General   
3                        B  75 3733 A Fleeing or Attempting To Elude Polic...   
4                        A  18 3921 A Theft by Unlawful Taking-Movable – L...   
5                        B          LO 6 101 Discharge of Firearms Prohibited   
..                     ...                                                ...   
96                       B  75 1501 A Drivers Required To Be Licensed—Gene...   
97                       B                         75 3736 A Reckless Driving   
98                       B    75 3309 1 Driving On Roadways Laned For Traffic   
99                       B  75 1543 B1.1I Driving While Operating Privileg...   
100                      B        75 1312 Notice of Change of Name or Address   

      XCOORD   YCOORD    Zone Tract     Neighborhood  \
_id               

**Now, we sort through the rows, to get a nice clean list of all of the neighborhoods that this dataset encompasses.**

In [3]:
crime = crime.dropna(subset=["Neighborhood"]).drop_duplicates()
neighborhoods = pd.DataFrame(sorted(crime["Neighborhood"].unique()), columns=["Neighborhoods"])

print(neighborhoods)
print("Neighborhoods covered:", len(neighborhoods))

          Neighborhoods
0      Allegheny Center
1        Allegheny West
2             Allentown
3             Arlington
4     Arlington Heights
..                  ...
86  Upper Lawrenceville
87             West End
88         West Oakland
89             Westwood
90              Windgap

[91 rows x 1 columns]
Neighborhoods covered: 91


**Now we can see that the dataset covers 91 different neighborhoods and their respective crimes.**

**Generally speaking, More Crime = Less Safety, and that metric works well enough for determining the safety of neighborhoods since crime types will usually average out. However, for vanity's sake, let's determine how many of each crime type have been committed throughout the various neighborhoods of Pittsburgh.**

In [10]:
crime_type = crime["Violation"]
crime_type.value_counts().head(10)

Violation
18 2701 A1 Simple Assault-General                                                                                                            2890
75 3745 A Accidents Involving Damage To Unattended Vehicle or Property                                                                       2825
75 3743 A Accidents Involving Damage To Attended Vehicle or Property                                                                         2307
18 3304 A1 Criminal Mischief- Destruction/Damage/Vandalism of Property- Basic Charge-$500 Or Less                                            2301
35 780-113 A32 CSDDCA-Use/Possession Drug Paraphernalia                                                                                      2027
18 3921 A Theft by Unlawful Taking-Movable - Motor Vehicle Theft-Autos                                                                       1838
18 2706 A1 Terroristic Threats-General                                                                            

**Here are some of the most "popular" crimes committed throughout Pittsburgh, and their respective counts. I am mildly concerned at the amount of terroristic threats, but otherwise they are mostly low grade, "standard" crimes.**

**Now, we'll count the amount of crimes in each neighborhood and sort them from most to least. Then we'll display the neighborhoods, from those with the least crimes to those with the most.**

In [11]:
crime_count = crime["Neighborhood"].value_counts().reset_index()
crime_count.columns = ["Neighborhoods", "Crime Count"]
crime_count = crime_count.sort_values(by="Crime Count", ascending=False)
crime_count.head(91)

Neighborhoods  Crime Count
0   Central Business District         6764
1            South Side Flats         4437
2                     Carrick         2135
3                   Shadyside         1913
4                East Liberty         1818
..                        ...          ...
86             Swisshelm Park           61
87              New Homestead           51
88                  Chartiers           38
89                  Ridgemont           32
90                Saint Clair           25

[91 rows x 2 columns]

**Here we can see that the most safe, and therefore "best" neighborhood, is Saint Clair, with a very low crime count of 25. On the other end of the spectrum, we have the Central Business District with a massive lead on the others, with a crime count of 6764. I would not want to live there. South Side Flats is also an honorable mention, as it too has a substantial lead from the neighborhoods behind (above?) it with a crime count of 4437